src/main/java/com/openshift/internal/restclient/DefaultClient.java

```diff
@@ -200,122 +162,104 @@ public String getResourceURI(IResource resource) {
 		
 	@Override
 	public <T extends IResource> T create(T resource, String namespace) {
-		return createVersion(resource, namespace, null);
+		switch(resource.getKind()) {
+		case ResourceKind.TEMPLATE:
+			return execute(HttpMethod.POST, ResourceKind.PROCESSED_TEMPLATES, namespace, null, null, resource);
+		default:
+			return execute(HttpMethod.POST, resource.getKind(), namespace, null, null, resource);
+		}
+	}
+
+	@Override
+	public <T extends IResource> T create(String kind, String namespace, String name, String subresource, IResource payload) {
+		return execute(HttpMethod.POST, kind, namespace, name, subresource, payload);
 	}
 	
-	private <T extends IResource> T createVersion(T resource, String namespace, String version) {
-		if(ResourceKind.LIST.equals(resource.getKind())) throw new UnsupportedOperationException("Generic create operation not supported for resource type 'List'");
-		try {
-			namespace = ResourceKind.PROJECT.equals(resource.getKind()) ? "" : namespace;
-			final URL endpoint = new URLBuilder(this.baseUrl, typeMapper)
-				.kind(resource.getKind())
+	enum HttpMethod{
+		GET,
+		PUT,
+		POST,
+		DELETE
+	}
+	
+	@SuppressWarnings("unchecked")
+	private <T extends IResource> T execute(HttpMethod method, String kind, String namespace, String name, String subresource, IResource payload) {
+		if(ResourceKind.LIST.equals(kind)) 
+			throw new UnsupportedOperationException("Generic create operation not supported for resource type 'List'");
+		namespace = ResourceKind.PROJECT.equals(kind) ? null : namespace;
```

Comment:
Removed line as the UrlBuilder along with the TypeMapper for apigroups handles this case